benchmark-service/benchmark/benchmark.py

```diff
@@ -2,19 +2,78 @@
 import aiohttp_jinja2
 from aiohttp import web
 import logging
-from gear import setup_aiohttp_session, web_authenticated_developers_only
+from gear import setup_aiohttp_session, web_authenticated_developers_only, check_csrf_token
 from hailtop.config import get_deploy_config
 from hailtop.tls import get_in_cluster_server_ssl_context
 from hailtop.hail_logging import AccessLogger, configure_logging
-from web_common import setup_aiohttp_jinja2, setup_common_static_routes
+from web_common import setup_aiohttp_jinja2, setup_common_static_routes, render_template
 
+import json
+import re
+from google.cloud import storage
+import plotly
+import plotly.express as px
 
 configure_logging()
 router = web.RouteTableDef()
 logging.basicConfig(level=logging.DEBUG)
 deploy_config = get_deploy_config()
 log = logging.getLogger('benchmark')
 
+FILE_PATH_REGEX = re.compile(r'((?P<user>[^/]+)/)((?P<version>[^-]+)-)((?P<sha>[^-]+))(-(?P<tag>[^\.]+))?\.json')
+filepath = 'tpoterba/0.2.45-ac6815ee857c-master.json'
+
+
+def parse_file_path(name):
+    match = FILE_PATH_REGEX.fullmatch(name)
+    return match.groupdict()
+
+
+def get_benchmarks(file_path):
+    # create storage client
+    storage_client = storage.Client()
+    # get bucket with name
+    bucket = storage_client.get_bucket('hail-benchmarks')
```

Comment:
This is great, but let's make a couple of changes.

1. Let's create a new file called `utils.py`. In it, let's create a class that represents reading from google storage. An attribute of the class should be the client. Then we want a method that reads an arbitrary file from google storage and returns the data as a string. You'll probably want to write a parser that figures out the bucket and then the rest of the path from a file path.